chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/downloadfile/GenerateListingDownloadFilesAspect.java

```diff
@@ -0,0 +1,155 @@
+
+        listingSetToJobNameMap.put(ListingSet.EDITION_2011, new DownloadJobRunInformation(DOWNLOAD_FILE_JOB_2011, env));
+        listingSetToJobNameMap.put(ListingSet.EDITION_2014, new DownloadJobRunInformation(DOWNLOAD_FILE_JOB_2014, env));
+        listingSetToJobNameMap.put(ListingSet.EDITION_2015, new DownloadJobRunInformation(DOWNLOAD_FILE_JOB_2015, env));
+    }
+
+    @AfterReturning("execution(* *.*(..)) && @annotation(generateListingDownloadFile)")
+    @Transactional
+    public void generateListingDownloadFile(JoinPoint joinPoint, GenerateListingDownloadFile generateListingDownloadFile) {
+        Arrays.asList(generateListingDownloadFile.listingSet()).stream()
+                .forEach(listingSet -> scheduleDownloadFileJob(listingSet));
+    }
+
+    private void scheduleDownloadFileJob(ListingSet listingSet) {
+        DownloadJobRunInformation info = listingSetToJobNameMap.get(listingSet);
+
+        if (!isJobAlreadyScheduled(info)) {
+            ChplOneTimeTrigger downloadFileTrigger = new ChplOneTimeTrigger();
+            ChplJob joinDevelopersJob = getDownloadFileJob(listingSetToJobNameMap.get(listingSet).jobName);
+            downloadFileTrigger.setJob(joinDevelopersJob);
+            downloadFileTrigger.setRunDateMillis(listingSetToJobNameMap.get(listingSet).getRunDateTime().toInstant().toEpochMilli());
+            downloadFileTrigger = addTriggerToScheduler(downloadFileTrigger);
+
+            LOGGER.info("System job {}/{} has been scheduled for {}",
+                    downloadFileTrigger.getJob().getGroup(),
+                    downloadFileTrigger.getJob().getName(),
+                    DateUtil.toLocalDateTime(downloadFileTrigger.getRunDateMillis()).toString());
+        }
+    }
+
+    private ChplOneTimeTrigger addTriggerToScheduler(ChplOneTimeTrigger trigger) {
+        try {
+            return schedulerManager.createBackgroundJobTrigger(trigger);
+        } catch (ValidationException | SchedulerException e) {
+            LOGGER.error("Could not schedule trigger - {}", trigger.toString(), e);
+            return null;
+        }
+    }
+
+    private ChplJob getDownloadFileJob(String jobName) {
+        return getAllJobs().stream()
+                .filter(job -> job.getName().equals(jobName))
+                .findAny()
+                .get();
+    }
+
+    private List<ChplJob> getAllJobs() {
+        try {
+            return schedulerManager.getAllJobs();
+        } catch (SchedulerException e) {
+            LOGGER.error("Could not retrieve list of Quartz jobs", e);
+            return List.of();
+        }
+    }
+
+    private boolean isJobAlreadyScheduled(DownloadJobRunInformation info) {
+        return getAllScheduledSystemJobs().stream()
+                .filter(systemJob -> systemJob.getName().equalsIgnoreCase(info.getJobName())
+                        && systemJob.getNextRunDate().getTime() == info.getRunDateTime().toInstant().toEpochMilli())
+                .findAny()
```

Comment:
I think this will always return "true" to "isAlreadyScheduled", as we have the 2011 and 2014 jobs in a "once per quarter" recurring schedule. I know we talked about removing that recurring schedule, but I don't think we can do that until we annotate all the other places that update the data in such a way that we'd want to regen those files. Is there a way to look for "isAlreadyScheduled SOON"?